scripts/pre-commit

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -e
+
+./scripts/check_versions.sh
+./scripts/check_lint.sh
+
+echo "CGO_ENABLED = $CGO_ENABLED"
+
+if [ "$CGO_ENABLED" -eq "0" ] || [ -z "$CGO_ENABLED" ]
+then
+    make test-norace
+else
+    make test
+fi
```

Comment:
@shan-96 could you clarify why do we check CGO here?

Reply:
The go test flag `-race` only works when C executables for Go are enabled. Some platforms might not even install while installing Go.

Reply:
@shan-96  I see, I didn't know..  on Mac it works without CGO starting from 1.20

> On macOS, the race detector has been rewritten not to use cgo: race-detector-enabled programs can be built and run without Xcode. On Linux and other Unix systems, and on Windows, a host C toolchain is required to use the race detector.